Packs/CiscoAMP/Integrations/CiscoAMP/CiscoAMP_test.py

```diff
@@ -0,0 +1,1153 @@
+"""
+Unit testing for CiscoAMP (Advanced Malware Protection)
+"""
+import json
+import io
+import os
+from typing import Dict, List, Any
+import pytest
+from CiscoAMP import Client
+
+API_KEY = 'API_Key'
+CLIENT_ID = 'Client_ID'
+SERVER_URL = 'https://api.eu.amp.cisco.com'
+BASE_URL = f'{SERVER_URL}/{Client.API_VERSION}'
+
+
+def assert_output_has_no_links(outputs: List[Dict]):
+    """
+    Check that there are no 'links' keys in the outputs.
+
+    Args:
+        outputs (List[Dict, str]): output to loop through.
+    """
+    for output in outputs:
+        assert 'links' not in output
+
+
+def load_mock_response(file_name: str) -> str | io.TextIOWrapper:
+    """
+    Load mock file that simulates an API response.
+    Args:
+        file_name (str): Name of the mock response JSON file to return.
+    Returns:
+        str: Mock file content.
+    """
+    path = os.path.join('test_data', file_name)
+
+    with io.open(path, mode='r', encoding='utf-8') as mock_file:
+        if os.path.splitext(file_name)[1] == '.json':
+            return json.loads(mock_file.read())
+
+        return mock_file
+
+
+@pytest.fixture(autouse=True)
+def mock_client() -> Client:
+    """
+    Establish a connection to the client with a URL and API key.
+
+    Returns:
+        Client: Connection to client.
+    """
+    from CommonServerPython import DBotScoreReliability
+
+    return Client(
+        server_url=SERVER_URL,
+        api_key=API_KEY,
+        client_id=CLIENT_ID,
+        reliability=DBotScoreReliability.C
+    )
+
+
+@pytest.mark.parametrize(
+    'args, suffix, file',
+    [({'limit': '34'}, '', 'computer_list_response.json'),
+     ({'connector_guid': '1'}, '/1', 'computer_get_response.json')]
+)
+def test_computer_list_command(requests_mock, mock_client, args, suffix, file):
+    """
+    Scenario:
+    -   Get a list of 34 computers.
+    -   Get a single computer.
+    Given:
+    -   The user has entered a limit.
+    -   The user has entered a connector_guid.
+    When:
+    -    cisco-amp-computer-list is called.
+    Then:
+    -   Ensure outputs_prefix is correct.
+    -   Ensure links don't exist.
+    """
+    mock_response = load_mock_response(file)
+    requests_mock.get(
+        f'{BASE_URL}/computers{suffix}',
+        json=mock_response
+    )
+
+    from CiscoAMP import computer_list_command
+    responses = computer_list_command(mock_client, args)
+
+    for response in responses[:-1]:
```

Comment:
why `[:-1]`?

Reply:
To not include the last `CommandResults`, which includes only a `readableoutput` for the command.